app-backend/database/src/main/scala/com/azavea/rf/database/tables/Datasources.scala

```diff
@@ -92,16 +93,25 @@ object Datasources extends TableQuery(tag => new Datasources(tag)) with LazyLogg
   /** Given a datasource ID, attempt to retrieve it from the database
     *
     * @param datasourceId UUID ID of datasource to get from database
+    * @param user         Results will be limited to user's organization
     */
-  def getDatasource(datasourceId: UUID) =
-    Datasources.filter(_.id === datasourceId).result.headOption
+  def getDatasource(datasourceId: UUID, user: User) =
+    Datasources
+      .filterToSharedOrganizationIfNotInRoot(user)
+      .filter(_.id === datasourceId)
```

Comment:
It does do the right thing. All filters are combined and translated into one SQL query under the hood by Slick. See more details here: http://slick.lightbend.com/doc/3.0.0/queries.html#sorting-and-filtering